### PR TITLE
[BE]: Apply FURB118 (prev): replaces unnecessary lambdas with operator.

### PR DESCRIPTION
**Pull request by Skylion007**
This replaces a bunch of unnecessary lambdas with the operator package. This is semantically equivalent, but the operator package is faster, and arguably more readable. When the FURB rules are taken out of preview, I will enable it as a ruff check.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: benchmarks/distributed/rpc/rl/agent.py [hunk 1]
```diff
@@ -75,9 +76,7 @@ def set_world(self, batch_size, state_size, nlayers, out_features, batch=True):
             batch (bool): Whether to process and respond to observer requests as a batch or 1 at a time
         """
         self.batch = batch
-        self.policy = Policy(
-            reduce((lambda x, y: x * y), state_size), nlayers, out_features
-        )
+        self.policy = Policy(reduce((operator.mul), state_size), nlayers, out_features)
```
Comment (malfet) [inline benchmarks/distributed/rpc/rl/agent.py:79]: Nit
```suggestion
        self.policy = Policy(reduce(operator.mul, state_size), nlayers, out_features)
```

File: benchmarks/tensorexpr/microbenchmarks.py [hunk 2]
```diff
@@ -105,10 +106,10 @@ def gen_float_comparison_tensors(N, M):
...
 te_bool = te.Dtype.Bool
 binary_ops = [
-    ("add", (lambda a, b: a + b), torch.add),
-    ("mul", (lambda a, b: a * b), torch.mul),
-    ("sub", (lambda a, b: a - b), torch.sub),
-    ("div", (lambda a, b: a / b), torch.div),
+    ("add", (operator.add), torch.add),
+    ("mul", (operator.mul), torch.mul),
+    ("sub", (operator.sub), torch.sub),
+    ("div", (operator.truediv), torch.div),
```
Comment (malfet) [inline benchmarks/tensorexpr/microbenchmarks.py:112]: Nit (I'm surprised we don't have lint for unneeded parentheses)
```suggestion
    ("add", operator.add, torch.add),
    ("mul", operator.mul, torch.mul),
    ("sub", operator.sub, torch.sub),
    ("div", operator.truediv, torch.div),
```

File: test/nn/test_pooling.py [hunk 3]
```diff
@@ -43,11 +44,11 @@ def _sum_pool3d(self, x, kernel_size):
         return joined_x.view(1, joined_x.numel())
 
     def _avg_pool2d(self, x, kernel_size):
-        size = reduce((lambda x, y: x * y), kernel_size)
+        size = reduce((operator.mul), kernel_size)
```
Comment (malfet) [inline test/nn/test_pooling.py:47]:
```suggestion
        size = reduce(operator.mul, kernel_size)
```

File: test/nn/test_pooling.py [hunk 4]
```diff
@@ -43,11 +44,11 @@ def _sum_pool3d(self, x, kernel_size):
...
         return self._sum_pool2d(x, kernel_size) / size
 
     def _avg_pool3d(self, x, kernel_size):
-        size = reduce((lambda x, y: x * y), kernel_size)
+        size = reduce((operator.mul), kernel_size)
```
Comment (malfet) [inline test/nn/test_pooling.py:51]:
```suggestion
        size = reduce(operator.mul, kernel_size)
```

File: torch/_utils.py [hunk 5]
```diff
@@ -494,7 +495,7 @@ def _import_dotted_name(name):
 
 
 # Taken from python 3.5 docs
-def _accumulate(iterable, fn=lambda x, y: x + y):
+def _accumulate(iterable, fn=operator.add):
```
Comment (malfet) [inline torch/_utils.py:498]: This is going to conflict with your other PR in a sec ;)

File: torch/ao/pruning/_experimental/data_sparsifier/data_norm_sparsifier.py [hunk 6]
```diff
@@ -35,7 +36,7 @@ def __init__(self, data_list: Optional[List[Tuple[str, Any]]] = None, sparsity_l
                  sparse_block_shape: Tuple[int, int] = (1, 4),
                  zeros_per_block: Optional[int] = None, norm: str = 'L1'):
         if zeros_per_block is None:
-            zeros_per_block = reduce((lambda x, y: x * y), sparse_block_shape)
+            zeros_per_block = reduce((operator.mul), sparse_block_shape)
```
Comment (malfet) [inline torch/ao/pruning/_experimental/data_sparsifier/data_norm_sparsifier.py:39]:
```suggestion
            zeros_per_block = reduce(operator.mul, sparse_block_shape)
```

File: torch/ao/pruning/_experimental/data_sparsifier/data_norm_sparsifier.py [hunk 7]
```diff
@@ -95,7 +96,7 @@ def __get_data_level_mask(self, data, sparsity_level,
         data_norm = F.avg_pool2d(data[None, None, :], kernel_size=sparse_block_shape,
                                  stride=sparse_block_shape, ceil_mode=True)
 
-        values_per_block = reduce((lambda x, y: x * y), sparse_block_shape)
+        values_per_block = reduce((operator.mul), sparse_block_shape)
```
Comment (malfet) [inline torch/ao/pruning/_experimental/data_sparsifier/data_norm_sparsifier.py:99]:
```suggestion
        values_per_block = reduce(operator.mul, sparse_block_shape)
```

File: torch/ao/pruning/_experimental/data_sparsifier/data_norm_sparsifier.py [hunk 8]
```diff
@@ -116,7 +117,7 @@ def __get_data_level_mask(self, data, sparsity_level,
     def update_mask(self, name, data, sparsity_level,
                     sparse_block_shape, zeros_per_block, **kwargs):
 
-        values_per_block = reduce((lambda x, y: x * y), sparse_block_shape)
+        values_per_block = reduce((operator.mul), sparse_block_shape)
```
Comment (malfet) [inline torch/ao/pruning/_experimental/data_sparsifier/data_norm_sparsifier.py:120]:
```suggestion
        values_per_block = reduce(operator.mul, sparse_block_shape)
```

File: torch/ao/pruning/sparsifier/weight_norm_sparsifier.py [hunk 9]
```diff
@@ -56,7 +57,7 @@ def __init__(self,
                  zeros_per_block: Optional[int] = None,
                  norm: Optional[Union[Callable, int]] = None):
         if zeros_per_block is None:
-            zeros_per_block = reduce((lambda x, y: x * y), sparse_block_shape)
+            zeros_per_block = reduce((operator.mul), sparse_block_shape)
```
Comment (malfet) [inline torch/ao/pruning/sparsifier/weight_norm_sparsifier.py:60]:
```suggestion
            zeros_per_block = reduce(operator.mul, sparse_block_shape)
```

File: torch/ao/pruning/sparsifier/weight_norm_sparsifier.py [hunk 10]
```diff
@@ -108,7 +109,7 @@ def _make_tensor_mask(self, data, input_shape, sparsity_level, sparse_block_shap
             mask.data = torch.ones_like(mask)
             return mask
 
-        values_per_block = reduce((lambda x, y: x * y), sparse_block_shape)
+        values_per_block = reduce((operator.mul), sparse_block_shape)
```
Comment (malfet) [inline torch/ao/pruning/sparsifier/weight_norm_sparsifier.py:112]:
```suggestion
        values_per_block = reduce(operator.mul, sparse_block_shape)
```

File: torch/ao/pruning/sparsifier/weight_norm_sparsifier.py [hunk 11]
```diff
@@ -174,7 +175,7 @@ def _make_block_mask(self, data, sparse_block_shape, zeros_per_block, mask=None)
 
     def update_mask(self, module, tensor_name, sparsity_level, sparse_block_shape,
                     zeros_per_block, **kwargs):
-        values_per_block = reduce((lambda x, y: x * y), sparse_block_shape)
+        values_per_block = reduce((operator.mul), sparse_block_shape)
```
Comment (malfet) [inline torch/ao/pruning/sparsifier/weight_norm_sparsifier.py:178]:
```suggestion
        values_per_block = reduce(operator.mul, sparse_block_shape)
```

File: torch/ao/pruning/sparsifier/weight_norm_sparsifier.py [hunk 12]
```diff
@@ -145,7 +146,7 @@ def _make_block_mask(self, data, sparse_block_shape, zeros_per_block, mask=None)
         block_h, block_w = sparse_block_shape
         dh = (block_h - h % block_h) % block_h
         dw = (block_w - w % block_w) % block_w
-        values_per_block = reduce((lambda x, y: x * y), sparse_block_shape)
+        values_per_block = reduce((operator.mul), sparse_block_shape)
```
Comment (malfet) [inline torch/ao/pruning/sparsifier/weight_norm_sparsifier.py:149]:
```suggestion
        values_per_block = reduce(operator.mul, sparse_block_shape)
```

**Review by malfet — APPROVED** [review]
LGTM, but please replace numerous `(operator.XYZ)` with `operator.XYZ`

File: torch/_utils.py (hunk 5 above)
Comment (Skylion007) [reply torch/_utils.py:498]: Yeah, I know. ;-;

**Pull request closed without merging**